python_modules/dagster/dagster_tests/daemon_tests/test_sensor_run.py

```diff
@@ -1499,3 +1520,54 @@ def test_run_status_sensor_interleave(external_repo_context, storage_config_fn):
                 )
                 assert len(ticks[0].origin_run_ids) == 1
                 assert ticks[0].origin_run_ids[0] == run1.run_id
+
+
+@pytest.mark.parametrize("external_repo_context", repos())
+def test_multi_job_sensor(external_repo_context):
+    freeze_datetime = to_timezone(
+        create_pendulum_time(year=2019, month=2, day=27, tz="UTC"),
+        "US/Central",
+    )
+    with instance_with_sensors(external_repo_context) as (
+        instance,
+        workspace,
+        external_repo,
+    ):
+        with pendulum.test(freeze_datetime):
+            job_sensor = external_repo.get_external_sensor("two_job_sensor")
+            instance.start_sensor(job_sensor)
+
+            evaluate_sensors(instance, workspace)
+
+            ticks = instance.get_job_ticks(job_sensor.get_external_origin_id())
+            assert len(ticks) == 1
+            validate_tick(
+                ticks[0],
+                job_sensor,
+                freeze_datetime,
+                JobTickStatus.SUCCESS,
+            )
+
+            run = instance.get_runs()[0]
+            assert run.run_config == {}
+            assert run.tags.get("dagster/sensor_name") == "two_job_sensor"
+            assert run.pipeline_name == "the_graph"
+
+            freeze_datetime = freeze_datetime.add(seconds=60)
+        with pendulum.test(freeze_datetime):
+
+            # should fire the asset sensor
+            evaluate_sensors(instance, workspace)
+            ticks = instance.get_job_ticks(job_sensor.get_external_origin_id())
+            assert len(ticks) == 2
+            validate_tick(
+                ticks[0],
+                job_sensor,
+                freeze_datetime,
+                JobTickStatus.SUCCESS,
+            )
+            run = instance.get_runs()[0]
+            assert run.run_config == {"solids": {"config_solid": {"config": {"foo": "blah"}}}}
+            assert run.tags
+            assert run.tags.get("dagster/sensor_name") == "two_job_sensor"
+            assert run.pipeline_name == "config_graph"
```

Comment:
It would be helpful to include tests that make sure we have a decent error message when:
* Someone includes a job name on a RunRequest for a single-job sensor
* Someone omits a job name on a RunRequest for a multi-job sensor, or supplies the name of a job that doesn't exist

Reply:
+1